modules/http2/h2_request.c

```diff
@@ -362,24 +371,43 @@ request_rec *h2_create_request_rec(const h2_request *req, conn_rec *c,
 #endif
 
 #if AP_MODULE_MAGIC_AT_LEAST(20120211, 107)
-    assign_headers(r, req, no_body);
+    assign_headers(r, req, no_body, is_connect);
     ap_run_pre_read_request(r, c);
 
     /* Time to populate r with the data we have. */
     r->request_time = req->request_time;
     AP_DEBUG_ASSERT(req->authority);
-    if (!apr_strnatcasecmp("CONNECT", req->method)) {
+    if (is_connect) {
       /* CONNECT MUST NOT have scheme or path */
+      if (req->scheme) {
+        ap_log_cerror(APLOG_MARK, APLOG_DEBUG, 0, c, APLOGNO()
+                      "':scheme: %s' header present in CONNECT request",
+                      req->scheme);
+        access_status = HTTP_BAD_REQUEST;
+        goto die;
+      }
+      if (req->path) {
+        ap_log_cerror(APLOG_MARK, APLOG_DEBUG, 0, c, APLOGNO()
+                      "':path: %s' header present in CONNECT request",
+                      req->path);
+        access_status = HTTP_BAD_REQUEST;
+        goto die;
+      }
       r->the_request = apr_psprintf(r->pool, "%s %s HTTP/2.0",
                                     req->method, req->authority);
     }
-    else if (req->scheme && ap_cstr_casecmp(req->scheme, "http")
-             && ap_cstr_casecmp(req->scheme, "https")) {
-        /* FIXME: we also need to create absolute uris when we are
-         * in a forward proxy configuration! But there is currently
-         * no way to detect that. */
+    else if (req->protocol) {
+      ap_log_cerror(APLOG_MARK, APLOG_DEBUG, 0, c, APLOGNO()
+                    "':protocol: %s' header present in %s request",
```

Comment:
Stray `'` in `%s'`?

Reply:
its the quote for the header, which is ':protocol: name'.